diffprivlib/models/tree.py

```diff
@@ -0,0 +1,602 @@
+
+        if not isinstance(self._cat_feature_threshold, numbers.Integral) or self._cat_feature_threshold < 0:
+            raise ValueError('Categorical feature threshold should be a positive integer;'
+                             f'got {self._cat_feature_threshold}')
+
+        self._accountant.check(self._epsilon, 0)
+
+        X, y = np.array(X), np.array(y)
+        check_X_y(X, y)
+
+        self._n_features = X.shape[1]
+        self._cat_features = get_cat_features(X, self._cat_feature_threshold)
+        self._max_depth = calc_tree_depth(n_cont_features=self._n_features-len(self._cat_features),
+                                          n_cat_features=len(self._cat_features), max_depth=self._max_depth)
+        self._classes = np.unique(y)
+        self._n_classes = len(self._classes)
+
+        if self._feature_domains is None:
+            warnings.warn(
+                "`feature_domains` parameter hasn't been specified, so falling back to determining domains from the data.\n"
+                "This may result in additional privacy leakage. To ensure differential privacy with no "
+                "additional privacy loss, specify `feature_domains` according to the documentation",
+                PrivacyLeakWarning)
+            self._feature_domains = get_feature_domains(X, self._cat_features)
+
+        if len(self._feature_domains) != self._n_features:
+            raise ValueError("Missing domains for some features in `feature_domains`")
+
+        if self._n_estimators > len(X):
+            raise ValueError('Number of estimators is more than the available samples')
+
+        subset_size = int(len(X) / self._n_estimators)
+        datasets = []
+        estimators = []
+
+        for i in range(self._n_estimators):
+            estimator = DecisionTreeClassifier(max_depth=self._max_depth, epsilon=self._epsilon, feature_domains=self._feature_domains)
+            estimator.set_cat_features(self._cat_features)
+            estimator.set_classes(self._classes)
+            estimators.append(estimator)
+            datasets.append(Dataset(X=X[i*subset_size:(i+1)*subset_size], y=y[i*subset_size:(i+1)*subset_size]))
+
+        estimators = Parallel(n_jobs=self._n_jobs, verbose=self._verbose, **_joblib_parallel_args(prefer='processes'))(
+            delayed(lambda estimator, X, y: estimator.fit(X, y))(estimator, dataset.X, dataset.y)
+            for estimator, dataset in zip(estimators, datasets)
+        )
+
+        self._estimators = estimators
+        self._accountant.spend(self._epsilon, 0)
+
+        return self
+
+    def check_is_fitted(self):
+        """Check if the model is fitted
+
+        Returns:
+            [bool]: True if the model is fitted, False otherwise.
+        """
+        if self._estimators:
+            for estimator in self._estimators:
+                if not estimator.check_is_fitted():
+                    return False
+            return True
+        return False
+
+    def predict(self, X):
```

Comment:
Let's try and get rid of this `predict()` method, since we can use the inherited method. This means we need a `predict_proba()` method in `DecisionTree()`. I'll investigate next week, but for our case this should mean returning a binary array with `1` at the index that we're predicting.